TLM/TLM/Custom/PathFinding/StockPathFind.cs

```diff
@@ -846,7 +844,15 @@ private struct BufferItem {
 						if (transitionNode) {
 							transitionCost *= 2f;
 						}
+						if (ticketCost != 0 && netManager.m_lanes.m_buffer[curLaneId].m_ticketCost != 0) {
+							transitionCost *= 10f;
+						}
 						float transitionCostOverMeanMaxSpeed = transitionCost / ((prevMaxSpeed + nextLaneInfo.m_speedLimit) * 0.5f * m_maxLength);
+						if (!this.m_stablePath && (netManager.m_lanes.m_buffer[curLaneId].m_flags & 0x80) != 0) {
+							int firstTarget = netManager.m_lanes.m_buffer[curLaneId].m_firstTarget;
+							int lastTarget = netManager.m_lanes.m_buffer[curLaneId].m_lastTarget;
+							transitionCostOverMeanMaxSpeed *= (float)new Randomizer(this.m_pathFindIndex ^ curLaneId).Int32(1000, (lastTarget - firstTarget + 2) * 1000) * 0.001f;
+						}
```

Comment:
This part was missing. Anyway, StockPathFind only exists to track changes of the stock PF class.